packages/feeds-client/src/feed/event-handlers/comment/handle-comment-added.ts

```diff
@@ -1,13 +1,33 @@
-import type { Feed } from '../../../feed';
-import type { EventPayload } from '../../../types-internal';
+import type { Feed } from '../../feed';
+import type { EventPayload, PartializeAllBut } from '../../../types-internal';
+import { getStateUpdateQueueId, shouldUpdateState } from '../../../utils';
+import { eventTriggeredByConnectedUser } from '../../../utils/event-triggered-by-connected-user';
+
+type CommentAddedPayload = PartializeAllBut<
+  EventPayload<'feeds.comment.added'>,
+  'comment'
+>;
 
 export function handleCommentAdded(
   this: Feed,
-  event: EventPayload<'feeds.comment.added'>,
+  payload: CommentAddedPayload,
+  fromWs?: boolean,
 ) {
-  const { comment } = event;
+  const { comment } = payload;
   const entityId = comment.parent_id ?? comment.object_id;
 
+  if (
+    !shouldUpdateState({
+      stateUpdateQueueId: getStateUpdateQueueId(comment, 'comment-added'),
+      stateUpdateQueue: this.stateUpdateQueue,
+      watch: this.currentState.watch,
+      fromWs,
+      isTriggeredByConnectedUser: eventTriggeredByConnectedUser.call(this, payload),
```

Comment:
nit: We don't have to use `this` function signature for everything, this one can accept a `Feed` normally - the `this` signature was for detached methods that can be used both in API call response handler and event handler forms while also keeping them private by not exporting them.

Reply:
^ it's the latter - I mostly use it for `protected` values.